### DISCUSSION TITLE
Disabling l_shift-l_alt key combinations

### DISCUSSION THREAD

**Discussion by janko-jj** [lxqt/lxqt#2623] [Q&A]
I'm attempting to "free" the l_shift - l_alt cursor movement key combinations to be able to use them inside of my editor.

I've removed the S-A-*  binds everywhere on my machine where I was able to find them (see the grep results included) (system: LUbuntu 22.04.5 LTS, LXQt Version: 0.17.0  Qt: 5.15.3  System Configuration: /etc/xdg).

But still: pressing Shift-Alt-SomeArrow behaves as initially, i.e. it cycles through the open windows?

How can I free these key combinations? I can't find that info? Thanks.

~~~
/etc/xdg$ grep -r "S-A" *
openbox/removed.xml:  <keybind key="S-A-Left">
openbox/removed.xml:  <keybind key="S-A-Right">
openbox/removed.xml:  <keybind key="S-A-Up">
openbox/removed.xml:  <keybind key="S-A-Down">
xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Left">
xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Right">
xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Up">
xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Down">
xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Left">
xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Right">
xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Up">
xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Down">
/etc/xdg$ cd ~/.config
~/.config$ grep -r "S-A" *
~/.config$ 
~~~

**Comment by stefonarch** [comment]
I don't remember exactly but it could be that you need to disable the shortcut setting it to nothing/empty, or you could use a if statement as below; otherwise the default will still be used.

```
<keybind key="C-S-Left">
  <action name="If">
    <title>* your app*</title>
    <then>
      <!-- Do nothing when this is the active window -->
    </then>
    <else>
      <!-- Do another thing in other cases-->
    </else>
  </action>
</keybind>
```
In labwc there is `<action name="None" />` but it doesn't look that it exists in openbox.

**Reply by janko-jj** [reply]
I've tried inserting that, both in /etc and locally, but I see no change. :(   As I just want to free the combinations, I've left the comments in both branches, like in your example.

**Reply by stefonarch** [reply]
You _do_ have the file ~/.config/openbox/rc.xml` ? I'm not sure if Lubuntu uses a different file, you may ask on their discord better. I logged a moment  in with openbox and S-A-up/down doesn't do anything here as I don't have any keybind configured, so if absent it should free them.. My default file in /etc/xdg/openbox says
```
  <keybind key="S-A-Left">
    <action name="SendToDesktop"><to>left</to><wrap>no</wrap></action>
  </keybind>
  <keybind key="S-A-Right">
    <action name="SendToDesktop"><to>right</to><wrap>no</wrap></action>
  </keybind>
  <keybind key="S-A-Up">
    <action name="SendToDesktop"><to>up</to><wrap>no</wrap></action>
  </keybind>
  <keybind key="S-A-Down">
    <action name="SendToDesktop"><to>down</to><wrap>no</wrap></action>
  </keybind>
```
while on the Lubuntu VM I see window switching without any rc.xml file configured. So maybe better ask on their side.

**Reply by janko-jj** [reply]
Thanks for trying to help. Here is where I've inserted your recipe, practically in every .xml that had such kind of entries, even if initially they weren't present:

~~~
~/.config/openbox$ grep "S-A" *
lubuntu-rc.xml:    <keybind key="S-A-Left">
lubuntu-rc.xml:  <keybind key="S-A-Right">
lubuntu-rc.xml:  <keybind key="S-A-Up">
lubuntu-rc.xml:  <keybind key="S-A-Down">
lxqt-rc.xml:  <keybind key="S-A-Left">
lxqt-rc.xml:  <keybind key="S-A-Right">
lxqt-rc.xml:  <keybind key="S-A-Up">
lxqt-rc.xml:  <keybind key="S-A-Down">
~/.config/openbox$ grep "S-A" /etc/xdg/xdg-Lubuntu/openbox/*
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="S-A-Left">
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="S-A-Right">
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="S-A-Up">
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="S-A-Down">
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Left">
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Right">
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Up">
/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml:  <keybind key="C-S-A-Down">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:    <keybind key="S-A-Left">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:  <keybind key="S-A-Right">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:  <keybind key="S-A-Up">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:  <keybind key="S-A-Down">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Left">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Right">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Up">
/etc/xdg/xdg-Lubuntu/openbox/rc.xml:  <keybind key="C-S-A-Down">
~/.config/openbox$ grep "S-A" /etc/xdg/openbox/*
/etc/xdg/openbox/lxqt-rc.xml:    <keybind key="S-A-Left">
/etc/xdg/openbox/lxqt-rc.xml:  <keybind key="S-A-Right">
/etc/xdg/openbox/lxqt-rc.xml:  <keybind key="S-A-Up">
/etc/xdg/openbox/lxqt-rc.xml:  <keybind key="S-A-Down">
/etc/xdg/openbox/rc.xml:  <keybind key="S-A-Left">
/etc/xdg/openbox/rc.xml:  <keybind key="S-A-Right">
/etc/xdg/openbox/rc.xml:  <keybind key="S-A-Up">
/etc/xdg/openbox/rc.xml:  <keybind key="S-A-Down">
~~~

Still after reboot I see only the windows switching as soon as I press e.g. S-A-up.

(Specifically: in the xmls which initially had C-S-A-variants there weren't present the S-A-variants, I've added them). Where the S-A- were present I've put these from you with "comments".

**Reply by stefonarch** [reply]
As I said: here on arch if it's not present it doesn't do anything. But window switching for S-A-left/right is not even the default behavior I think. Anyway only rc.xml in your home are counting. You can change the theme with `obconf-qt` and check for the date to understand which one is used by Lubuntu.

Btw there is an official ppa to update to LXQt 1.4

**Reply by janko-jj** [reply]
"only rc.xml in your home" "counting" 

Many thanks! That was the info I missed! I saw all these .xml files and had no idea which is actually active (and haven't tried the theme thing this time). But I've recognized that I haven't had the "rc.xml" in the ~/.config/openbox,  just lubuntu-rc.xml and lxqt-rc.xml there (I thought they do something if they are there -- that the distro renamed the "default" conf filename or something like that) and changing them changed nothing (and strangely, also changing the /etc/xdg/openbox/rc.xml  changed nothing(?)). But after I've copied lubuntu-rc.xml to rc.xml in my .config/openbox, and _completely removed the offending keybinds_ I finally have the combinations working in my editor!

**Reply by stefonarch** [reply]
I was about to ask if you have that file... lxqt-rc.xml is from very old versions and not in use anymore. It's not the file in /etc reading,  the default shortcuts are in the code.

 I changed this to Q&A.

**Reply by janko-jj** [reply]
Wait, do I understand correctly, the /etc ... files are _never_ read anymore? That should be clearly written everywhere then.

**Reply by stefonarch** [reply]
They are there for reference, often they will  be copied when creating a new user, depends on the program. You can use the lxqt-panel without config, it will have it's defaults. I saw in my Lubuntu installation the same, no rc.xml in ~/.config/openbox/` until you change a value.

**Reply by janko-jj** [reply]
OK, so can this be a full summary: openbox has 

1) the defaults hardcoded in the binaries, which aren't the same on all distros.

2) it reads the ~/.config/openbox/rc.xml  for the user

3) lxqt-rc.xml is name that is not used anymore by openbox at all.

Any rc.xml anywhere else, even in /etc, is not read by openbox on startup, it is there maybe just to be used by whatever external entity as a kind of template?
